libcodechecker/libhandlers/server.py

```diff
@@ -446,6 +679,70 @@ def main(args):
     check_env = analyzer_env.get_check_env(context.path_env_extra,
                                            context.ld_lib_path_extra)
 
+    cfg_sql_server = database.SQLServer.from_cmdline_args(
+        vars(args), CONFIG_META, context.config_migration_root,
+        interactive=True, env=check_env)
+
+    LOG.info("Checking configuration database ...")
+    db_status = cfg_sql_server.connect()
+    db_status_msg = database_status.db_status_msg.get(db_status)
+    LOG.info(db_status_msg)
+
+    if db_status == DBStatus.SCHEMA_MISSING:
+        LOG.debug("Config database schema is missing, initializing new.")
+        db_status = cfg_sql_server.connect(init=True)
+        if db_status != DBStatus.OK:
+            LOG.error("Config database initialization failed!")
+            LOG.error("Please check debug logs.")
+            sys.exit(1)
+
+    if db_status == DBStatus.SCHEMA_MISMATCH_NO:
+        LOG.debug("Configuration database schema mismatch.")
+        LOG.debug("No schema upgrade is possible.")
+        sys.exit(1)
+
+    if db_status == DBStatus.SCHEMA_MISMATCH_OK:
+        LOG.debug("Configuration database schema mismatch.")
+        LOG.debug("Schema upgrade is possible.")
+        LOG.warning("Please note after migration only "
+                    "newer CodeChecker versions can be used"
+                    "to start the server")
+        LOG.warning("It is advised to make a full backup of your "
+                    "configuration database")
+        if util.get_user_input(question):
```

Comment:
`question` is undefined here.

 1. Create default config database in SQLite.
 2. Stop server.
 3. Create a random migration script for config db
 4. Start server.